### PR TITLE
Feature - add dependency tracking with SQL connection string

### LINKED ISSUE #65: Provide capability to track a SQL dependency based on connection string information
**Is your feature request related to a problem? Please describe.**
Provide capability to track a SQL dependency based on connection string information instead of having to pass server name et all ourselves.

**Describe the solution you'd like**
`_logger.LogSqlDependency(connectionString, "my-table", "get-products", isSuccessful: true, startTime: startTime, duration: durationMeasurement.Elapsed, context: telemetryContext);`

**Describe alternatives you've considered**
None.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #65

### CODE REVIEW

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 1]
```diff
@@ -301,6 +302,42 @@ public static void LogHttpDependency(this ILogger logger, HttpRequestMessage req
...
+        /// <param name="logger">Logger to use</param>
+        /// <param name="connectionString">SQL connection string</param>
+        /// <param name="tableName">Name of table</param>
+        /// <param name="operationName">Name of the operation that was performed</param>
+        /// <param name="startTime">Point in time when the interaction with the HTTP dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogSqlDependency(this ILogger logger, string connectionString,  string tableName, string operationName, DateTimeOffset startTime, TimeSpan duration,  bool isSuccessful, Dictionary<string, object> context = null)
+        {
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:335]: Should we guard for empty connection string or does the builder handle that?

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 2]
```diff
@@ -301,6 +302,42 @@ public static void LogHttpDependency(this ILogger logger, HttpRequestMessage req
...
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="connectionString">SQL connection string</param>
+        /// <param name="tableName">Name of table</param>
+        /// <param name="operationName">Name of the operation that was performed</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the SQL dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogSqlDependency(this ILogger logger, string connectionString,  string tableName, string operationName, DependencyMeasurement measurement, bool isSuccessful, Dictionary<string, object> context = null)
+        {
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:316]: Should we guard for empty connection string or does the builder handle that?

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:335]: Hmm, the connection string builder doesn't guard against null or empty. Maybe we should handle that.

Comment (tomkerkhove) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:335]: Think that's a good idea!

Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:335]: Done.

**Pull request merged**